lib/ruby_lsp/requests/base_request.rb

```diff
@@ -79,6 +79,15 @@ def locate_node_and_parent(parent, target_nodes, position)
           []
         end
       end
+
+      sig { params(node: T.nilable(SyntaxTree::Node), range: T.nilable(T::Range[Integer])).returns(T::Boolean) }
+      def visible?(node, range)
+        return true if range.nil?
+        return false if node.nil?
+
+        loc = node.location
+        range.cover?(loc.start_line - 1) && range.cover?(loc.end_line - 1)
```

Comment:
I think something like `SyntaxTree::Location#in?(range)` for this would be handy 🤔